newton/_src/sim/builder.py

```diff
@@ -194,6 +194,20 @@ class ShapeConfig:
         If provided (and sdf_target_voxel_size is None), enables SDF-based mesh-mesh collision.
         Set to None (default) to disable SDF generation for this shape (uses BVH-based collision for mesh-mesh instead).
         Requires GPU since wp.Volume only supports CUDA. Only used for mesh shapes."""
+        is_hydroelastic: bool = False
+        """Whether the shape collides using SDF-based hydroelastics. For hydroelastic collisions, both participating shapes must have is_hydroelastic set to True. Defaults to False.
+
+        .. note::
+            Hydroelastic collision handling only works with volumetric shapes.
+            This flag will be set to False for planes and heightfields in :meth:`ModelBuilder.add_shape`.
```

Comment:
Does it make sense to make it clear that this list is not exhaustive? E.g. a flat mesh for cloth? And why do we set to false, would not the solver just ignore if it's not supported?